dspace-api/src/main/java/org/dspace/content/service/RelationshipService.java

```diff
@@ -160,5 +226,47 @@ Relationship create(Context c, Item leftItem, Item rightItem, RelationshipType r
      */
     Relationship create(Context c, Item leftItem, Item rightItem, RelationshipType relationshipType,
                         int leftPlace, int rightPlace)
-            throws AuthorizeException, SQLException;
+        throws AuthorizeException, SQLException;
+
+    /**
+     * counts all relationships
+     *
+     * @param context DSpace context object
+     * @return total relationships
+     * @throws SQLException if database error
+     */
+    int countTotal(Context context) throws SQLException;
+
+    /**
+     * Count total number of relationships (rows in relationship table) by a relationship type
+     *
+     * @param context context
+     * @param relationshipType relationship type to filter by
+     * @return total count
+     * @throws SQLException if database error
+     */
+    int countByRelationshipType(Context context, RelationshipType relationshipType) throws SQLException;
+
+    /**
+     * This method returns a count of Relationship objects that have the given Item object
+     * as a leftItem or a rightItem
+     * @param context   The relevant DSpace context
+     * @param item      The item that should be either a leftItem or a rightItem of all
+     *                  the Relationship objects in the returned list
+     * @return          The list of Relationship objects that contain either a left or a
+     *                  right item that is equal to the given item
+     * @throws SQLException If something goes wrong
+     */
+    int countByItem(Context context, Item item) throws SQLException;
+
+    /**
+     * Count total number of relationships (rows in relationship table) by a relationship type
+     *
+     * @param context context
+     * @param relationshipType relationship type to filter by
+     * @return total count
+     * @throws SQLException if database error
+     */
+    int countByItemAndRelationshipType(Context context, RelationshipType relationshipType, Item item)
```

Comment:
This should be:
```java
int countByItemAndRelationshipType(Context context, Item item, RelationshipType relationshipType)
```

Reply:
Corrected:
https://github.com/DSpace/DSpace/pull/2535/commits/240a142d9b9ae7e5fc4a22b5ad92637497b5ea11#diff-d205ffe807bd29f516efd140431ba927R270